internal/parser/translate_httproute.go

```diff
@@ -43,67 +43,99 @@ func fromHTTPRoutes(log logrus.FieldLogger, httpRouteList []*gatewayv1alpha2.HTT
 			// Therefore we treat each match rule as a separate Kong Route, so we iterate through
 			// all matches to determine all the routes that will be needed for the services.
 			var routes []kongstate.Route
-			for matchNumber, match := range rule.Matches {
-				// determine the name of the route, identify it as a route that belongs
-				// to a Kubernetes HTTPRoute object.
-				routeName := kong.String(fmt.Sprintf(
-					"httproute.%s.%s.%d",
-					httproute.Namespace,
-					httproute.Name,
-					matchNumber, // TODO: avoid route thrash from re-ordering?
-				))
-
-				// TODO: implement query param matches
-				if len(match.QueryParams) > 0 {
-					errmsg := "query param matches are not yet supported"
-					log.Errorf("HTTPRoute %s/%s can't be routed for match %+v: %s", errmsg)
-					continue
-				}
+			if len(rule.Matches) > 0 {
+				for matchNumber, match := range rule.Matches {
+					// determine the name of the route, identify it as a route that belongs
+					// to a Kubernetes HTTPRoute object.
+					routeName := kong.String(fmt.Sprintf(
+						"httproute.%s.%s.%d",
+						httproute.Namespace,
+						httproute.Name,
+						matchNumber, // TODO: avoid route thrash from re-ordering?
+					))
+
+					// TODO: implement query param matches
+					if len(match.QueryParams) > 0 {
+						errmsg := "query param matches are not yet supported"
+						log.Errorf("HTTPRoute %s/%s can't be routed for match %+v: %s", errmsg)
+						continue
+					}
 
-				// TODO: implement regex path matches
-				if *match.Path.Type == gatewayv1alpha2.PathMatchRegularExpression {
-					errmsg := "regular expression path matches are not yet supported"
-					log.Errorf("HTTPRoute %s/%s can't be routed for match %+v: %s", errmsg)
-					continue
-				}
+					// TODO: implement regex path matches
+					if *match.Path.Type == gatewayv1alpha2.PathMatchRegularExpression {
+						errmsg := "regular expression path matches are not yet supported"
+						log.Errorf("HTTPRoute %s/%s can't be routed for match %+v: %s", errmsg)
```

Comment:
this repetition of errorf with a shared message prefix is an indication of a code smell. The translator for an individual route should be a function returning `error`, and then the error can be wrapped in an error of the form `HTTPRoute %s/%s can't be routed for match %+v`.

Reply:
It's certainly a problem that's endemic to this area of the code. I've wrapped it as you suggested for this function in df6143bd400d21d5f7b9178067220d0e3ed5d903, but noted that there are still several problems with how this area of the code works and left a comment to https://github.com/Kong/kubernetes-ingress-controller/issues/2130 for future followup.

Reply:
looks very good :+1: thanks